Libplanet.RocksDBStore/RocksDBStore.cs

```diff
@@ -1361,25 +1361,47 @@ private void LogUnexpectedException(string methodName, Exception e)
             bool includeDeleted
         )
         {
+            long count = 0;
+            Stack<(Guid, long)> chainInfos = new Stack<(Guid, long)>();
+
             if (!includeDeleted && IsDeletionMarked(chainId))
             {
                 yield break;
             }
 
-            long count = 0;
+            chainInfos.Push((chainId, CountIndex(chainId)));
 
-            if (GetPreviousChainInfo(chainId) is { } chainInfo)
+            while (true)
             {
-                Guid prevId = chainInfo.Item1;
-                long pi = chainInfo.Item2;
-
-                int expectedCount = (int)(pi - offset + 1);
-                if (limit is { } limitNotNull && limitNotNull < expectedCount)
+                if (chainInfos.Peek().Item2 > offset &&
+                    GetPreviousChainInfo(chainInfos.Peek().Item1) is { } chainInfo)
                 {
-                    expectedCount = limitNotNull;
+                    chainInfos.Push(chainInfo);
                 }
+                else
+                {
+                    break;
+                }
+            }
+
+            long previousChainTipIndex;
 
-                foreach (BlockHash hash in IterateIndexes(prevId, offset, expectedCount, true))
+            // Adjust offset if it skipped some previous chains.
+            (offset, previousChainTipIndex) =
+                GetPreviousChainInfo(chainInfos.Peek().Item1) is { } cinfo
+                    ? (Math.Max(0, (int)(offset - cinfo.Item2 - 1)), (int)cinfo.Item2)
```

Comment:
Is this casting (i.e., `(int)`) safe? I guess `(long)` seems natural... (of course, it is unnatural that `offset` was `int`, but... 🤔 )

Reply:
Yes, I agree about it. In my thought, it was `int` because LINQ `Enumerable.Skip` supports only `int` type as argument 😭 So... it may be better to skip manually (with `continue` in `for-loop`)

Reply:
I made it handle block indexes as `long` type instead of `int` type in [cdc8b2a](https://github.com/planetarium/libplanet/pull/2379/commits/cdc8b2a2d2daa1fb191786286d7ffa3f3935dba6). Please review again 🙇🏻‍♂️